src/resources/views/crud/columns/inc/bulk_actions_checkbox.blade.php

```diff
@@ -53,9 +53,17 @@ function addOrRemoveCrudCheckedItem(element) {
 
     if (typeof markCheckboxAsCheckedIfPreviouslySelected !== 'function') {
         function markCheckboxAsCheckedIfPreviouslySelected() {
+            let checkedItems = [];
             document
                 .querySelectorAll('input.crud_bulk_actions_line_checkbox[data-primary-key-value]')
-                .forEach(elem => elem.checked = crud.checkedItems?.length && crud.checkedItems.indexOf(elem.dataset.primaryKeyValue) > -1);
+                .forEach(function(elem) {
+                    let checked = crud.checkedItems?.length && crud.checkedItems.indexOf(elem.dataset.primaryKeyValue) > -1;
+                    elem.checked = checked;
+                    if(checked) {
+                        checkedItems.push(elem.dataset.primaryKeyValue);
+                    }
```

Comment:
I think this would work better if we rewrite it to use X:

```suggestion
                    let checked = crud.checkedItems?.length && crud.checkedItems.indexOf(elem.dataset.primaryKeyValue) > -1;
                    elem.checked = checked;
                    if (checked) { checkedItems.push(elem.dataset.primaryKeyValue); }
```